students/barecki_andrzej/lesson_10_organizing_files/selective_copy.py

```diff
@@ -0,0 +1,63 @@
+import os
+import datetime
+import shutil
+
+SRC_FOLDER = "source_folder"
+DEST_FOLDER = "destination_folder"
+
+
+def create_example_file_tree(path, source_folder_param, des_folder_param):
+    os.chdir(path)
+    print('Current folder ', os.getcwd())
+    src_dir = ""
+
+    """Create source folder(extend date/time approach)"""
+    if source_folder_param is not None:
+        if os.path.exists(source_folder_param):
+            random_folder = os.path.join(source_folder_param + '_' +
+                                         str(datetime.datetime.now().date()) +
+                                         '_' + str(datetime.datetime.now().
+                                                   time()).replace(':', '.'))
+            src_dir = random_folder
+        else:
+            src_dir = source_folder_param
+
+    """ Create example of few files """
+    file_list = [os.path.join(src_dir, "dir_1", "test1.jpg"),
+                 os.path.join(src_dir, "dir_1", "test2.pdf"),
+                 os.path.join(src_dir, "dir_1", "test3.doc"),
+                 os.path.join(src_dir, "dir_1", "dir_2", "test4.jpg"),
+                 os.path.join(src_dir, "dir_1", "dir_2", "test5.pdf"),
+                 os.path.join(src_dir, "dir_1", "dir_2", "test6.doc"),
+                 os.path.join(src_dir, "dir_1", "dir_2", "dir_3", "test7.jpg"),
+                 os.path.join(src_dir, "dir_1", "dir_2", "dir_3", "test8.pdf"),
+                 os.path.join(src_dir, "dir_1", "dir_2", "dir_3", "test9.doc"),
+                 ]
+    for elem in file_list:
```

Comment:
Podoba mi się jak starasz się rozdzielać dane i algorytmy. Programy wtedy samoistnie zaczynają dzielić się na sekcje deklaracji i sekcje wykonywania operacji. Ludzki mózg dobrze reaguje na taki podział, bo ma duże obszary deklaracji gdzie nie trzeba myśleć i mniejsze obszary wykonywania obliczeń, gdzie trzeba się skupić. To sprawia, że kod czyta się łatwo. W przypadku gdy nie ma takiego podziału, trzeba utrzymywać wysokie skupienie cały czas, bo w każdej linijce można oczekiwać istotnych operacji.

Spójrz jak można jeszcze bardziej to uprościć, żeby uniknąć wielokrotnego powtarzania `os.path.join(src_dir`:
```
file_list = [["dir_1", "test1.jpg"],
             ["dir_1", "test2.pdf"],
             ["dir_1", "dir_2", "dir_3", "test9.doc"],
          ]

print([os.path.join(src_dir, *path) for path in file_list])
```